pkg/repository/v1_repository.go

```diff
@@ -551,6 +554,16 @@ func (r *V1Repository) FetchIndex() (index *v1manifest.Index, err error) {
 	return index, nil
 }
 
+// DownloadTiup downloads the tiup tarball and expands it into targetDir
+func (r *V1Repository) DownloadTiup(targetDir string) error {
+	var spec = ComponentSpec{
```

Comment:
The `targetDir` useless? I think you forget to assign the `targetDir` to `ComponentSpec.TargetDir`.

Reply:
fixed in https://github.com/pingcap-incubator/tiup/pull/257/commits/11beea8feb1616ed3125ea23ab942913d25b45f0